src/TicketingSystem.EventsApi/Controllers/EventsController.cs

```diff
@@ -0,0 +1,115 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using TicketingSystem.BusinessLogic.Dtos;
+using TicketingSystem.BusinessLogic.Enums;
+using TicketingSystem.BusinessLogic.Services;
+using TicketingSystem.EventsApi.Models;
+
+namespace TicketingSystem.EventsApi.Controllers
+{
+    /// <summary>
+    /// API to manipulate over Events, EventSections, EventRows (inline entity), EventSeats, Tickets
+    /// </summary>
+    [ApiController]
+    [Route("[controller]")]
+    public class EventsController : ControllerBase
+    {
+        private readonly IEventService _eventService;
+        private readonly IEventSeatService _eventSeatService;
+        private readonly IEventSectionService _eventSectionService;
+        private readonly IMapper _mapper;
+
+        public EventsController(
+            IEventService eventService,
+            IEventSeatService eventSeatService,
+            IEventSectionService eventSectionService,
+            IMapper mapper)
+        {
+            _eventService = eventService;
+            _eventSeatService = eventSeatService;
+            _eventSectionService = eventSectionService;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Returns a list of Venues with their info (no additional entities attached)
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetEvents()
+        {
+            var events = await _eventService.GetAllAsync();
+
+            return Ok(events);
+        }
+
+        /// <summary>
+        /// Returns a list of Event Seats (with Event Section id, Event Row Id) with their statuses and prices
+        /// </summary>
+        /// <param name="eventId"></param>
+        /// <param name="sectionId"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{eventId}/sections/{sectionId}/seats")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetFullSeats([FromRoute] string eventId, [FromRoute] string sectionId)
+        {
```

Comment:
There is too many logic in this endpoint. I would suggest creating for example EventSeatService (name doesn't matter), inject all necessary repositories there, and do all this DB calls and validations. Then your controllers will be just clear calls to services without validation and mapping logic. But this is up to you.

Reply:
Moved extra logic from controllers to services